daemon/src/close_position/taker.rs

```diff
@@ -0,0 +1,106 @@
+use crate::close_position::protocol::*;
+use crate::command;
+use anyhow::Context;
+use anyhow::Result;
+use async_trait::async_trait;
+use model::libp2p::PeerId;
+use model::OrderId;
+use model::Price;
+use tokio_tasks::Tasks;
+use xtra::Address;
+use xtra_libp2p::Endpoint;
+use xtra_productivity::xtra_productivity;
+
+pub struct Actor {
+    endpoint: Address<Endpoint>,
+    tasks: Tasks,
+    executor: command::Executor,
+    n_payouts: usize,
+}
+
+impl Actor {
+    pub fn new(endpoint: Address<Endpoint>, executor: command::Executor, n_payouts: usize) -> Self {
+        Self {
+            endpoint,
+            tasks: Tasks::default(),
+            executor,
+            n_payouts,
+        }
+    }
+}
+
+#[async_trait]
+impl xtra::Actor for Actor {
+    type Stop = ();
+
+    async fn stopped(self) -> Self::Stop {}
+}
+
+#[derive(Clone, Copy)]
+pub struct ClosePosition {
+    pub order_id: OrderId,
+    pub price: Price,
+    pub maker_peer_id: PeerId,
+}
+
+#[xtra_productivity]
+impl Actor {
+    pub async fn handle(&mut self, msg: ClosePosition) -> Result<()> {
+        let ClosePosition {
+            order_id,
+            price,
+            maker_peer_id,
+        } = msg;
+
+        let (close_position_tx, _) = self
+            .executor
+            .execute(order_id, |cfd| {
+                cfd.start_close_position_taker(price, self.n_payouts)
+            })
+            .await
+            .context("could not start closing position")?;
```

Comment:
Probably the idea is that this error should be reported back to the caller. I am not sure I agree though - unless the caller actually does something sensible with this I would move it into the task. reporting back information only makes sense if the caller does something with it. If the caller is not interested we should design the API accordingly.

Reply:
we do this because we're checking for invariants (ie. whether we are able to perform a collab close). If we check here, we actually show this is the taker's UI, which gives good user experience.

It's preserving current behaviour, and I'm reluctant to change within this PR.
@da-kami I believe it's the caller's decision whether to care or not and we can't tell from this point of code.